engine/gui/src/test/test_gui.cpp

```diff
@@ -1817,33 +1817,48 @@ TEST_F(dmGuiTest, ScriptAnimate)
 
 TEST_F(dmGuiTest, ScriptPlayback)
 {
-    dmGui::HNode node = dmGui::NewNode(m_Scene, Point3(0,0,0), Vector3(10,10,0), dmGui::NODE_TYPE_BOX);
-    dmGui::SetNodeId(m_Scene, node, "n");
-    const char* s = "function init(self)\n"
+    dmGui::Result r;
+    int durations[9] = { 1, 1,  1, 0, 0,  0, -1, -1, -1 };
+    int delays[9]     = { 1, 0, -1, 1, 0, -1,  0,  1, -1 };
+    const char* s = "function cb(self, node)\n"
+                    "    assert(self.foobar == 123)\n"
+                    "    gui.set_color(node, vmath.vector4(1.0,0,0,0))\n"
+                    "end\n;"
+                    "function init(self)\n"
+                    "    self.foobar = 123\n"
                     "    self.node = gui.get_node(\"n\")\n"
-                    "    gui.animate(self.node, gui.PROP_POSITION, vmath.vector4(1,0,0,0), gui.EASING_NONE, 1, 0, nil, gui.PLAYBACK_ONCE_BACKWARD)\n"
-                    "end\n"
-                    "function final(self)\n"
-                    "    gui.delete_node(self.node)\n"
+                    "    gui.animate(self.node, gui.PROP_POSITION, vmath.vector4(1,0,0,0), gui.EASING_NONE, %d, %d, cb, gui.PLAYBACK_ONCE_BACKWARD)\n"
                     "end\n";
+    for (int i = 0; i < 9; ++i)
+    {
+        dmGui::HNode node = dmGui::NewNode(m_Scene, Point3(0,0,0), Vector3(10,10,0), dmGui::NODE_TYPE_BOX);
+        dmGui::SetNodeId(m_Scene, node, "n");
+        char buffer[1024];
 
-    dmGui::Result r;
-    r = dmGui::SetScript(m_Script, LuaSourceFromStr(s));
-    ASSERT_EQ(dmGui::RESULT_OK, r);
-
-    r = dmGui::InitScene(m_Scene);
-    ASSERT_EQ(dmGui::RESULT_OK, r);
+        sprintf(buffer, s, durations[i], delays[i]);
```

Comment:
We use `dmSNPrintf(buffer, sizeof(buffer), s, durations[i], delays[i])` from "slib/dstrings.h"